lib/internal/asyncEachOfLimit.js

```diff
@@ -0,0 +1,59 @@
+const breakLoop = require('./breakLoop')
+
+// for async generators
+export default function asyncEachOfLimit(generator, limit, iteratee, callback) {
+    let done = false
+    let canceled = false
+    let awaiting = false
+    let running = 0
+    let idx = 0
+
+    function replenish() {
+        //console.log('replenish')
+        if (running >= limit || awaiting) return
+        //console.log('replenish awaiting')
+        awaiting = true
+        generator.next().then(({value, done: iterDone}) => {
```

Comment:
Similar to what @megawac was saying, I think we need a `if (done) return;` line at the very top of this callback. If one of the previous iteratees cancelled or errored, we shouldn't be invoking the` iteratee` with successive items.

For example, if the first `iteratee` starts and we start waiting for the second item. If the first `iteratee` cancels before the `then` callback is called with the second item, when it is called, we would still invoke the `iteratee` with the second item.